### PR TITLE
JSON RPC null field updates

### PR DESCRIPTION
**Pull request by tgerring**
Placeholder per #884 

Updates to `eth_getBlockByNumber` and `eth_getTransactionByHash` complete. Changes to filter logs blocked

### CODE REVIEW

**Comment by zelig** [comment]
not 100% if its the same but a contract transaction shows up like this:
```
> eth.getBlock("pending", true).transactions;
[{
  to: Error: ,
...
}]
```

also see related #1006

File: rpc/api.go [hunk 1]
```diff
@@ -240,9 +246,12 @@ func (api *EthereumApi) GetRequestReply(req *RpcRequest, reply *interface{}) err
 		tx, bhash, bnum, txi := api.xeth().EthTransactionByHash(args.Hash)
 		if tx != nil {
 			v := NewTransactionRes(tx)
-			v.BlockHash = newHexData(bhash)
-			v.BlockNumber = newHexNum(bnum)
-			v.TxIndex = newHexNum(txi)
+			// if the blockhash is 0, assume this is a pending transaction
+			if bytes.Compare(bhash.Bytes(), []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}) != 0 {
```
Comment (Gustav-Simonsson) [inline rpc/api.go]: http://golang.org/pkg/bytes/#Repeat

**Comment by tgerring** [comment]
I see no error when using curl:
```
$ curl --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["pending", true],"id":1}' http://127.0.0.1:8545
{
	"id": 1,
	"jsonrpc": "2.0",
	"result": {
		"number": null,
		"hash": null,
		"parentHash": "0xba903f302eac4d8ab452d2f159e75ed2a866bdb158ff5913308728a41c93afdd",
		"nonce": null,
		"sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
		"logsBloom": null,
		"transactionsRoot": "0xf83e5cfbd16f8b6c91ee8c5ff7249f37dc020986443963c3d63d4e4a17f7a083",
		"stateRoot": "0x58120a84ff13921af725e1919782cedb90337abaa1fb682ea09e1cbc22149e2b",
		"miner": null,
		"difficulty": "0x2580e968b",
		"totalDifficulty": "0x0",
		"size": "0x2e1",
		"extraData": "0x",
		"gasLimit": "0x2fefd8",
		"gasUsed": "0x0",
		"timestamp": "0x555b3230",
		"transactions": [
			{
				"hash": "0x3889f2ba6c183986f140bc993df466923c8d60a366d80ecb6e6093395b34a92b",
				"nonce": "0x3b7c",
				"blockHash": "0x76b98c46a5a8360cfb1dd966ec0d7c99c3e1925044b186342360a492272d51eb",
				"blockNumber": "0x5de79",
				"transactionIndex": "0x0",
				"from": "0x4759966877052299c659d84023a144d244aaf42c",
				"to": "0x5a219cab7d5a5888ae86e80de2775dc022de6e0c",
				"value": "0x0",
				"gas": "0x76c00",
				"gasPrice": "0x9184e72a000",
				"input": "0xe25fe175"
			},
			{
				"hash": "0xe01198379ebe304192aaa24d9a32e467ed8d16ca14ca0c74cb222deb2d8f4bf3",
				"nonce": "0x3b7d",
				"blockHash": "0x76b98c46a5a8360cfb1dd966ec0d7c99c3e1925044b186342360a492272d51eb",
				"blockNumber": "0x5de79",
				"transactionIndex": "0x1",
				"from": "0x4759966877052299c659d84023a144d244aaf42c",
				"to": "0x5a219cab7d5a5888ae86e80de2775dc022de6e0c",
				"value": "0x0",
				"gas": "0x76c00",
				"gasPrice": "0x9184e72a000",
				"input": "0xe25fe175"
			}
		],
		"uncles": []
	}
}
```

**Comment by frozeman** [comment]
you need to have contact creation tx, one without a to

**Comment by tgerring** [comment]
If someone comes up with a minimum viable test case to demonstrate that this is a problem in the RPC and not the JavaScript library on top (as the issue was originally raised through JS), I can apply a fix. IT must be something that I can reproduce on the command-line with curl--I won't layer browsers or any extraneous software on top when testing.

**Comment by debris** [comment]
That's a problem with [pp_js.go](https://github.com/ethereum/go-ethereum/blob/develop/jsre/pp_js.go). It's not printing nulls.

File: xeth/xeth.go [hunk 2]
```diff
@@ -869,6 +871,7 @@ func (self *XEth) Transact(fromStr, toStr, nonceStr, valueStr, gasStr, gasPriceS
 		contractCreation bool
 	)
 
+	// 2015-05-18 Is this still needed?
```
Comment (obscuren) [inline xeth/xeth.go]: Eventually. I'll open a PR for this when I have some time.

**Pull request merged**